pkg/clusterapi/comparator.go

```diff
@@ -0,0 +1,27 @@
+package clusterapi
+
+import (
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+)
+
+// Comparator holds CAPI control plane and worker objects for comparisons.
+type Comparator[C Object[C], M Object[M]] struct {
+	ControlPlane[C, M]
+	Workers[M]
+}
+
+// KubernetesVersionModified checks if kubernetes version differs from current state KubeadmControlPlane.
+func KubernetesVersionModified(oldKubeadm controlplanev1.KubeadmControlPlane, newKubeadm controlplanev1.KubeadmControlPlane) bool {
+	return oldKubeadm.Spec.Version != newKubeadm.Spec.Version
+}
+
+// ControlPlaneScaleUpDown returns the change in number of replicas of kubeadm.
+func ControlPlaneScaleUpDown(oldKubeadm controlplanev1.KubeadmControlPlane, newKubeadm controlplanev1.KubeadmControlPlane) int32 {
```

Comment:
any reason not return bool?

Reply:
I think the number of replicas to scale up/down will be needed for later Tinkerbell scaling upgrade hardware validations and I wanted to make this reusable for that.

Reply:
I think this should return 0 if there is no scale up/down, and then a positive or negative integer depending on whether we are scaling up or down - I was hoping this could be used for the assertion later to calculate the number of hardware we need. 

Should I make another function/replace this one with one that returns bool for use with IsRollingNotScale()?

Reply:
No longer using bool to track operation.